src/nbl/video/IPhysicalDevice.cpp

```diff
@@ -103,58 +103,226 @@ inline core::bitflag<asset::IImage::E_ASPECT_FLAGS> getImageAspects(asset::E_FOR
+    case asset::EF_ASTC_4x4_SRGB_BLOCK:
+    case asset::EF_ASTC_5x4_UNORM_BLOCK:
+    case asset::EF_ASTC_5x4_SRGB_BLOCK:
+    case asset::EF_ASTC_5x5_UNORM_BLOCK:
+    case asset::EF_ASTC_5x5_SRGB_BLOCK:
+    case asset::EF_ASTC_6x5_UNORM_BLOCK:
+    case asset::EF_ASTC_6x5_SRGB_BLOCK:
+    case asset::EF_ASTC_6x6_UNORM_BLOCK:
+    case asset::EF_ASTC_6x6_SRGB_BLOCK:
+    case asset::EF_ASTC_8x5_UNORM_BLOCK:
+    case asset::EF_ASTC_8x5_SRGB_BLOCK:
+    case asset::EF_ASTC_8x6_UNORM_BLOCK:
+    case asset::EF_ASTC_8x6_SRGB_BLOCK:
+    case asset::EF_ASTC_8x8_UNORM_BLOCK:
+    case asset::EF_ASTC_8x8_SRGB_BLOCK:
+    case asset::EF_ASTC_10x5_UNORM_BLOCK:
+    case asset::EF_ASTC_10x5_SRGB_BLOCK:
+    case asset::EF_ASTC_10x6_UNORM_BLOCK:
+    case asset::EF_ASTC_10x6_SRGB_BLOCK:
+    case asset::EF_ASTC_10x8_UNORM_BLOCK:
+    case asset::EF_ASTC_10x8_SRGB_BLOCK:
+    case asset::EF_ASTC_10x10_UNORM_BLOCK:
+    case asset::EF_ASTC_10x10_SRGB_BLOCK:
+    case asset::EF_ASTC_12x10_UNORM_BLOCK:
+    case asset::EF_ASTC_12x10_SRGB_BLOCK:
+    case asset::EF_ASTC_12x12_UNORM_BLOCK:
+    case asset::EF_ASTC_12x12_SRGB_BLOCK:
+        // (All of these could be using HDR. Take extra flag to assume FP16 precision?)
+        rcpUnit = 1.0 / 255.0;
+        break;
+    case asset::EF_EAC_R11_UNORM_BLOCK:
+    case asset::EF_EAC_R11_SNORM_BLOCK:
+    case asset::EF_EAC_R11G11_UNORM_BLOCK:
+    case asset::EF_EAC_R11G11_SNORM_BLOCK:
+        rcpUnit = 1.0 / 2047.0;
+        break;
+    case asset::EF_ETC2_R8G8B8_UNORM_BLOCK:
+    case asset::EF_ETC2_R8G8B8_SRGB_BLOCK:
+    case asset::EF_ETC2_R8G8B8A1_UNORM_BLOCK:
+    case asset::EF_ETC2_R8G8B8A1_SRGB_BLOCK:
+    case asset::EF_ETC2_R8G8B8A8_UNORM_BLOCK:
+    case asset::EF_ETC2_R8G8B8A8_SRGB_BLOCK:
+        rcpUnit = 1.0 / 31.0;
+        break;
+    case asset::EF_BC6H_UFLOAT_BLOCK:
+    case asset::EF_BC6H_SFLOAT_BLOCK:
+    {
+        // BC6 isn't really FP16, so this is an over-estimation
+        float f = 0.0;
+        uint16_t f16 = core::Float16Compressor::compress(f);
+        uint16_t dir = core::Float16Compressor::compress(2.f * (f + 1.f));
+        return core::Float16Compressor::decompress(core::nextafter16(f16, dir)) - f;
+    }
+    case asset::EF_PVRTC1_2BPP_UNORM_BLOCK_IMG:
+    case asset::EF_PVRTC1_4BPP_UNORM_BLOCK_IMG:
+    case asset::EF_PVRTC2_2BPP_UNORM_BLOCK_IMG:
+    case asset::EF_PVRTC2_4BPP_UNORM_BLOCK_IMG:
+    case asset::EF_PVRTC1_2BPP_SRGB_BLOCK_IMG:
+    case asset::EF_PVRTC1_4BPP_SRGB_BLOCK_IMG:
+    case asset::EF_PVRTC2_2BPP_SRGB_BLOCK_IMG:
+    case asset::EF_PVRTC2_4BPP_SRGB_BLOCK_IMG:
+        // TODO
+        // I couldn't find any documentation on how this works
+        return 0.0;
+    }
 
-    for (uint32_t i = 0; i < channels; i++)
+    if (isSRGBFormat(format))
     {
-        min = std::min(min, asset::getFormatPrecision<float>(f, i, 0.0));
+        return core::srgb2lin(0.0 + rcpUnit) - core::srgb2lin(0.0);
     }
 
-    return min;
+    return rcpUnit;
```

Comment:
I think you're messing up by reporting only the precision of the end-points (as seen in your BC1-5 cases)

Remember that the endpoint itself can be interpolated with interpolants.

Imagine an endpoint at `(0,0,0)` and `(nextUlpRed,nextUlpGreen,nextUlpBlue)`

Despite the fact that the endpoints can only be placed at 1/32 or 1/64 intervals, a single pixel has an interpolant which can interpolate the endpoints (in BC, IIRC its 2 bits, so your max precision is 1/128 or 1/256)